### DISCUSSION TITLE
Disable WiFi on an ethernet capable Zigbee bridge

### DISCUSSION THREAD

**Discussion by bushwoelie** [arendst/Tasmota#21122] [Q&A] [answered]
I have a Zigbee bridge configured for ZHA that is connected via ethernet to my home network. It came preloaded with Tasmota firmware. At the moment it still has it's open WiFi AP enabled so you can configure the WiFi network it has to connect to if you get it new out of the box (which is an obvious security risk just leaving it this way).
I just want to disable the WiFi of the unit because I'm not going to use it. I cannot find that option anywhere in the Tasmota webinterface.

**Comment by sfromis** [comment, marked as answer]
Here's a small piece of sample Berry code which will disable WiFi shortly after boot, if Ethernet is active. It also can reopen Wifi as a fallback in case of Ethernet not operational.
https://tasmota.github.io/docs/Berry-Cookbook/#ethernet-network-flipper
That code also shows what command can be used.

Note that only a pretty small part of Tasmota features are in the quite rudimentary web UI, nearly everything is controlled by [commands](https://tasmota.github.io/docs/Commands/).

**Reply by bushwoelie** [reply]
Ok, I'm not quite familiar with the advanced functions of Tasmota but I found the "Manage File System", created a "autoexec.be" file and added the code from you link. Rebooted the device and still the SSID of the AP is available.
After that opened the Berry Console and ran the function netflip() manually. Still the AP available and no error in the console.
Ran netflip2() to check what happens and then get a "undeclared function" error so the autoexec.be I created is loaded and the netflip() function runs without errors but does not disable the WiFi.

Found it! The script does not work when the unit is still in AP mode. I had to connect it to a WiFi network before WiFi could be disabled.

Thanks a lot for the info!

**Reply by sfromis** [reply]
When you have no AP configured, Tasmota will start in Wifi Manager AP mode, expecting you to configure the AP to connect to. And at that stage, Tasmota has no STA mode IP address, meaning that the network flipper code does not see Wifi as being present, and will not issue the command `Wifi 0` which it would do if it saw Wifi as as operational.

Of course, the code could easily be changed to accommodate this, but as already hinted, the code also shows the command which can be used to disable Wifi. Also mentioned above....

(You edited while I prepared the response, but never mind)

**Comment by Jason2866** [comment]
Imho the WiFi AP should always be opened (as it is now). If things go wrong device is not reachable anymore....
